lib/middleware/anonymous_cache.rb

```diff
@@ -32,6 +33,23 @@ def blocked_crawler?
         CrawlerDetection.is_blocked_crawler?(@env[USER_AGENT])
       end
 
+      def is_ie=(val)
+        @is_ie = val ? :true : :false
```

Comment:
Why are we using symbols for `true` and `false` here when we could use a boolean?

If it's because you want to memoize it, you can use https://jtway.co/how-to-memoize-false-and-nil-values-fe28b8ede9f8 `defined?`

Reply:
@eviltrout I'm using the patterns already established in this library:

https://github.com/discourse/discourse/blob/master/lib/middleware/anonymous_cache.rb#L36

Reply:
@jjaffeux @SamSaffron has agreed to migrate away from that pattern in favor of `defined?`